src/components/todo/TodoItem.tsx

```diff
@@ -16,25 +20,84 @@ const TodoItem = ({ todoItems, setTodos, index }: Props) => {
     setTodos(todos);
   };
 
+  // State to check edit state(if already in edit mode)
+  const [editMode, setEditMode] = useState<boolean>(false);
+  //state to store edit value
+  const [editTodo, setEditTodo] = useState<string>(todoItem.todo);
+  //state to store date value
+  const [editDate, setEditDate] = useState<Date>(todoItem.date);
+
   // function to handel delete task
   const handleTaskDelete = (id: number) => {
     setTodos(todoItems.filter((todoItem) => todoItem.id !== id));
   };
 
+  // function to handel Edit icon // passing id //
+  const handleTaskEdit = (e: React.FormEvent, id: number, index: number) => {
+    e.preventDefault();
+    let todos = [...todoItems];
+    if (todos[index]) {
+      todos[index].todo = editTodo;
+      todos[index].date = editDate;
+    }
+    setTodos(todos);
+    setEditMode(false);
+  };
+
+  // function to handel enter key in edit task textbox
+  const editTask = (e: any) => {
+    console.log("User pressed: ", e.key);
+    if (e.key === "Enter") {
+      handleTaskEdit(e, todoItem.id, index);
+    }
+  };
+
+  const defaultDate = new Date(editDate).toISOString().split("T")[0]; // yyyy-mm-dd
   return (
     <div className="list-item">
-      <span
-        className={`list-item-text 
+      {editMode ? (
+        //If in edit mode then display Textbox
+        <div className="edit-task-container">
+          <input
```

Comment:
We can use https://react-hook-form.com/ for forms

Reply:
Updated using react-hook-form